aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -157,15 +157,13 @@ void copy_cast_mps(at::Tensor& dst, const at::Tensor& src,
       }
     }
 
-    size_t size_to_copy = src.nbytes();
+    size_t size_to_copy = dst.nbytes();
```

Comment:
This is needed for .item() calls. which copy just 1 element.
E.g if src is 12 bytes and dst is 4 bytes, we should copy only 4 bytes from src (not the whole 12)

Reply:
But that doesn't answer this question: What if dst.nbytes() > src.nbytes()?
Should we do this instead?
```suggestion
    size_t size_to_copy = std::min(src.nbytes(), dst.nbytes());
```

Reply:
This would happen only if the data types are not the same - this is handled at https://github.com/kulinseth/pytorch/blob/master/aten/src/ATen/native/mps/operations/Copy.mm#L149

Reply:
Still not sure why would it hurt to make the copy fail-safe with `std::min(src.nbytes(), dst.nbytes())`, but that's OK if you think it wouldn't crash in future.

Reply:
Right, this seems a bit risky. @DenisVieriu97 , can you update the scalar local dense as we discussed and skip changing the Copy code.

Reply:
Can we move it back to src.nbytes() change ?

Reply:
Updated